### ISSUE TITLE
Assistant messages with empty content cause: ValidationException - text content blocks must be non-empty 

### ISSUE THREAD

**Issue by ms-ati** [patterns-ai-core/langchainrb#896]
**Describe the bug**
Using `langchainrb` with `AwsBedrock`, we are blocked by **ValidationException** due to empty `content` in a message:

  * `/ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/seahorse/client/plugins/raise_response_errors.rb:17:in 'call': messages: text content blocks must be non-empty (Aws::BedrockRuntime::Errors::ValidationException)`
  
Why? After the LLM **uses a Tool**, we intermittently observe _Assistant Messages_ coming back with empty content, like this:

  * `#<Langchain::Assistant::Messages::AnthropicMessage:0x00007f6a27ba5250 @role="assistant", @content="", @image_url=nil, @tool_calls=[{"type"=>"tool_use", "id"=>"toolu_bdrk_01Eha5werhJV2mvQpWojtGxm", "name"=>"intervention_strategies_provider__intervention_strategies", "input"=>{}}], @tool_call_id=nil>`

I found a similar issue filed in `langchainjs`:
https://github.com/langchain-ai/langchainjs/issues/5960

**Backtrace**

```
.gem/ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call': messages: text content blocks must be non-empty (Aws::BedrockRuntime::Errors::ValidationException)
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:16:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/invocation_id.rb:16:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/seahorse/client/plugins/request_callback.rb:89:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/seahorse/client/plugins/response_target.rb:24:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/telemetry.rb:39:in `block in call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/telemetry/no_op.rb:29:in `in_span'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/telemetry.rb:53:in `span_wrapper'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/aws-sdk-core/plugins/telemetry.rb:39:in `call'
        from /ruby/3.3.2/gems/aws-sdk-core-3.214.0/lib/seahorse/client/request.rb:72:in `send_request'
        from /ruby/3.3.2/gems/aws-sdk-bedrockruntime-1.34.0/lib/aws-sdk-bedrockruntime/client.rb:1925:in `invoke_model'
        from /ruby/3.3.2/gems/langchainrb-0.19.2/lib/langchain/llm/aws_bedrock.rb:155:in `chat'
        from /ruby/3.3.2/gems/langchainrb-0.19.2/lib/langchain/assistant.rb:348:in `chat_with_llm'
        ... 10 levels...
```

**To Reproduce**
Use `Langchain::LLM::AwsBedrock` using Tools, and observe that some _Assistant_ messages will appear with empty `content` -- but only sometimes, not all the time.

Once such an empty-content message has been added to the `@messages` list, no further interactions will pass validation, causing the AI Agent to be in a non-usable state.

**Expected behavior**
Expect that _Assistant_ messages after Tool use will never have empty `content`, therefore never triggering this error state.

**Desktop (please complete the following information):**

- OS: Mac OS Sonoma 14.7.1
- Ruby version: 3.3.2
- Langchain.rb version: 0.19.2

**Additional context**

Similar issue filed with the project `langchainjs`: https://github.com/langchain-ai/langchainjs/issues/5960

**Comment by ms-ati** [comment]
Hi @andreibondarev, appreciate your help. I am using `Langchain::LLM::AwsBedrock` without specifying the model explicitly, so I am assuming I am getting the [defaults](https://github.com/patterns-ai-core/langchainrb/blob/9bbd609c689904a584817fff35a057e7868bee34/lib/langchain/llm/aws_bedrock.rb#L13-L23)

```ruby
    DEFAULTS = {
      chat_model: "anthropic.claude-3-5-sonnet-20240620-v1:0",
      completion_model: "anthropic.claude-v2:1",
      embedding_model: "amazon.titan-embed-text-v1",
      max_tokens_to_sample: 300,
      temperature: 1,
      top_k: 250,
      top_p: 0.999,
      stop_sequences: ["\n\nHuman:"],
      return_likelihoods: "NONE"
    }.freeze
```

**Comment by ms-ati** [comment]
I am currently making only the following override to the options:

```ruby
    default_options: {
      max_tokens_to_sample: 8_192
    }
```

**Comment by andreibondarev** [comment]
@ms-ati I went ahead and merged the fix. Would you mind testing it out?

**Comment by hakunin** [comment]
Been getting this a lot today since Anthropic seems to be under heavy load and tool uses produce no content with them.
Had to resort to this monkeypatch (but it's put together hastily)

```ruby
module Langchain
  class Assistant
    module LLM
      module Adapters
        class Anthropic < Base
          def build_chat_params(messages:, instructions:, tools:, tool_choice:, parallel_tool_calls:)
            messages = messages
            messages = messages.map do |msg|
              msg = msg.with_indifferent_access
              if msg[:content]&.empty?
                msg.merge(content: "(blank)")
              elsif msg[:content].is_a?(Array)
                msg.merge(content: msg[:content].map { |c| c["text"]&.empty? ? c.merge("text" => "(blank)") : c })
              else
                msg
              end
            end
            
            params = {messages: messages}
            if tools.any?
              params[:tools] = build_tools(tools)
              params[:tool_choice] = build_tool_choice(tool_choice, parallel_tool_calls)
            end
            params[:system] = instructions if instructions
            params
          end
        end
      end
    end
  end
end
```

**Comment by ms-ati** [comment]
Any chance of a release that includes https://github.com/patterns-ai-core/langchainrb/commit/41e232b31f96dee414db65d0b6b0819faca83955 ?

**Comment by andreibondarev** [comment]
> Any chance of a release that includes [41e232b](https://github.com/patterns-ai-core/langchainrb/commit/41e232b31f96dee414db65d0b6b0819faca83955) ?

Just released `0.19.3`.
